x/ccv/provider/handler_test.go

```diff
@@ -101,7 +105,31 @@ func TestAssignConsensusKeyForConsumerChain(t *testing.T) {
 			chainID:  "chainid",
 		},
 		{
-			name: "fail: consumer key in use",
+			name: "fail: consumer key in use by other validator",
+			setup: func(ctx sdk.Context,
+				k keeper.Keeper, mocks testkeeper.MockedKeepers,
+			) {
+				k.SetPendingConsumerAdditionProp(ctx, &providertypes.ConsumerAdditionProposal{
+					ChainId: "chainid",
+				})
+				// Use the consumer key already
+				k.SetValidatorByConsumerAddr(ctx, "chainid", consumerConsAddr, providerConsAddr2)
+
+				gomock.InOrder(
+					mocks.MockStakingKeeper.EXPECT().GetValidator(
+						ctx, providerCryptoId.SDKValOpAddress(),
+						// Return a valid validator, found!
+					).Return(providerCryptoId2.SDKStakingValidator(), true).Times(1),
```

Comment:
Why is this returning `providerCryptoId2` and not `providerCyrptoId`?
My understanding is that you set  `k.SetValidatorByConsumerAddr(ctx, "chainid", consumerConsAddr, providerConsAddr2)` and then when we try to set `consumerConsAddr` for `providerConsAddr` this test would fail, correct?

Reply:
You are right, fixed, thanks. It also ran into an error with the previous behaviour, so the test didn't catch the difference :/